File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let failIdx = 0;
    for(let i = 1; i < words.length; i++){
        let val = words[i];
        let prevVal = words[i-1];
        if(prevVal.substring(prevVal.length - 1) !== val.substring(0, 1)) {
            failIdx = i;
            break;
        } 
        if(words.indexOf(val) !== i) {
            failIdx = i;
            break;
        }
    }

    if(failIdx == 0) return [0,0];

    let people = failIdx % n + 1;
    let turn = Math.floor(failIdx / n) + 1; 

    return [people, turn];
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    used_words = set()
    prev_word = words[0][0]
    
    for i, word in enumerate(words):
        if word in used_words or word[0] != prev_word:
            return [(i % n) + 1, (i // n) + 1]
        used_words.add(word)
        prev_word = word[-1]
    return [0, 0]